src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -865,6 +863,49 @@ public LogStream logs(final String containerId, final LogsParameter... params)
   }
 
   @Override
+  public EventStream events(EventsParam... params)
+          throws DockerException, InterruptedException {
+    WebTarget resource = noTimeoutResource()
+            .path("events");
+    final Map<String, String> filters = newHashMap();
+    for (EventsParam param : params) {
+      if (param instanceof EventsFilterParam) {
+        filters.put(param.name(), param.value());
+      } else {
+        resource = resource.queryParam(param.name(), param.value());
+      }
+    }
+
+    try {
+      if (!filters.isEmpty()) {
+        final StringWriter writer = new StringWriter();
+        final JsonGenerator generator = objectMapper().getFactory().createGenerator(writer);
+        generator.writeStartObject();
+        for (Map.Entry<String, String> entry : filters.entrySet()) {
+          generator.writeArrayFieldStart(entry.getKey());
+          generator.writeString(entry.getValue());
+          generator.writeEndArray();
+        }
+        generator.writeEndObject();
+        generator.close();
+        // We must URL encode the string, otherwise Jersey chokes on the double-quotes in the json.
+        final String encoded = URLEncoder.encode(writer.toString(), UTF_8.name());
+        resource = resource.queryParam("filters", encoded);
+      }
+    } catch (IOException exception) {
+      throw new DockerException(exception);
+    }
+
+    try {
+      CloseableHttpClient client = (CloseableHttpClient) ApacheConnectorProvider.getHttpClient(noTimeoutClient);
```

Comment:
This cast is safe. For whatever reason, ApacheConnectorProvider.getHttpClient downcasts to HttpClient despite its own internal property being a ClosableHttpClient